### PR TITLE
Fix unique name validation

### PR DESCRIPTION
**Pull request by timchurchard**

### CODE REVIEW

File: pkg/register/document.go [hunk 1]
```diff
@@ -122,24 +121,24 @@ func (document RegisterDocument) Validate() []error {
 	}
 
 	// Check key/delegation names are unique in the document
-	bufOfNames := ""
+	bufOfNames := map[string]bool{}
```
Comment (StephanieBracaloni) [inline pkg/register/document.go:124]: :+1:

File: pkg/register/document.go [hunk 2]
```diff
@@ -122,24 +121,24 @@ func (document RegisterDocument) Validate() []error {
...
 
 	for _, publicKey := range document.PublicKeys {
 		errs = append(errs, validatePublicKey(PublicKeyTypeString, publicKey)...)
 
-		if strings.Contains(bufOfNames, publicKey.ID+"|") {
+		if _, found := bufOfNames[publicKey.ID]; found {
 			errs = append(errs, fmt.Errorf("key name '%s' is not unique", publicKey.ID))
 		}
-		bufOfNames = bufOfNames + publicKey.ID + "|"
+		bufOfNames[publicKey.ID] = true
```
Comment (StephanieBracaloni) [inline pkg/register/document.go:132]: (we could add a continue in the if to avoid updating the map if already present)

**Review by StephanieBracaloni — APPROVED** [review]
:+1:

File: pkg/register/document.go (hunk 2 above)
Comment (timchurchard) [reply pkg/register/document.go:133]: Thanks, I've added the continues.

**Pull request merged**